### PR TITLE
publish bgs mock container

### PR DESCRIPTION
**Pull request by msnwatson**
## What was the problem?

In order to run integration tests in the remote environments, we need to have the mock service running alongside the application before we do the work understand the data that is available in lower environments. 

Associated tickets or Slack threads:
- #2999 

## How does this fix it?[^1]

* Begins publishing the mock BGS container and running it through SecRel for future deployments alongside testing for BGS

## How to test this PR
- Step 1
- Step 2[^secrel]

[^1]: [Pull-Requests guidelines](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Pull-Requests). If PR is significant, update [Current Software State](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Current-Software-State) wiki page.
[^secrel]: To check if a PR will succeed in the SecRel workflow, [test PRs in the SecRel pipeline](https://github.com/department-of-veterans-affairs/abd-vro-internal/wiki/Secure-Release-process#to-test-prs-in-the-secrel-pipeline).

### CODE REVIEW

**Comment by msnwatson** [comment]
To reviewers: checks that are failing were already failing prior to these changes.

File: helm/svc-bgs-api/templates/tests/integration-tests.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+apiVersion: v1
+kind: Pod
+metadata:
+  name: "{{ include "demo.fullname" . }}-test-connection"
+  labels:
+    {{- include "demo.labels" . | nindent 4 }}
```
Comment (lisac) [inline helm/svc-bgs-api/templates/tests/integration-tests.yaml:6]:
```suggestion
  name: vro-svc-bgs-api-test
  labels: {{- toYaml .Values.labels | nindent 4 }}
```

File: helm/svc-bgs-api/templates/tests/integration-tests.yaml [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  annotations:
+    "helm.sh/hook": test
+spec:
+  containers:
+      - name: svc-bgs-api{{ include "vro.containerSuffix" . }}-test
+        image: { { include "vro.imageRegistryPath" . } }vro-svc-bgs-api:{{ include "vro.imageTag" . }}
```
Comment (lisac) [inline helm/svc-bgs-api/templates/tests/integration-tests.yaml]:
```suggestion
        image: {{ include "vro.imageRegistryPath" . }}vro-svc-bgs-api:{{ include "vro.imageTag" . }}
```

File: helm/svc-bgs-api/templates/tests/integration-tests.yaml (hunk 1 above)
Comment (lisac) [inline helm/svc-bgs-api/templates/tests/integration-tests.yaml]:
```suggestion
  name: vro-svc-bgs-api-test
  labels: {{- toYaml .Values.labels | nindent 4 }}
```

**Review by lisac — APPROVED** [review]
a minor suggestion on the helm chart to address `helm lint` flags, otherwise looks good to me.

with that suggestion in-place locally, a `helm lint` showed no failures. (what i used: `helm lint  helm/svc-bgs-api    --debug  -f helm/_shared/values.yaml -f helm/_shared/values-for-dev.yaml -f helm/svc-bgs-api/values.yaml`)

File: helm/svc-bgs-api/templates/tests/integration-tests.yaml (hunk 2 above)
Comment (msnwatson) [reply helm/svc-bgs-api/templates/tests/integration-tests.yaml]: ah i actually wanted to put this into a separate PR

**Pull request merged**